core/src/identity.rs

```diff
@@ -1,15 +1,28 @@
 use serde::{Deserialize, Serialize};
 
-#[derive(serde::Serialize, serde::Deserialize, Clone, Debug)]
-pub struct KeyMaterial {
-    pub id: String,
-    #[serde(rename = "type")]
-    pub proof_type: String,
-    pub controller: String,
-    pub public_key_multibase: String,
+#[derive(Clone, Debug, Copy)]
```

Comment:
Can you delete this and have `recover` accept the `resolver` via parameters similar to `generate`? I believe we discussed the reasons why we should prefer having these functions be "pure" and lifecycle-less as part of the last PR